src/main/java/org/skriptlang/skript/scheduler/Task.java

```diff
@@ -0,0 +1,140 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.scheduler;
+
+import org.bukkit.plugin.Plugin;
+
+import ch.njol.skript.Skript;
+import ch.njol.util.Closeable;
+
+/**
+ * Represents a Task that only calls it's runnable synchronous.
+ * Unless otherwise defined by the scheduler like Folia.
+ * <p>
+ * See {@link AsyncTask} for asynchronous tasks.
+ */
+public abstract class Task implements Runnable, Closeable {
+
+	private final Plugin plugin;
+	private long periodInTicks = -1;
+
+	public Task(long delayInTicks, long periodInTicks) {
+		this(Skript.getInstance(), delayInTicks, periodInTicks);
+	}
+
+	public Task(Plugin plugin, long delayInTicks, long periodInTicks) {
+		this.plugin = plugin;
+		this.periodInTicks = periodInTicks;
+		TaskManager.run(this, delayInTicks);
```

Comment:
I'm a little concerned with both this and AsyncTask that they're scheduled as soon as they're created. Can this be moved to a run method instead?